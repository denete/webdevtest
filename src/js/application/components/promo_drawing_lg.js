import React from "react";
import { Link } from "react-router";
import dateFormat from "dateformat"

// large format for drawing schedule table
export default class PromoDrawingLg extends React.Component {
  render(){
    const drawings = this.props.drawings;

    const rows = drawings.map((o,i)=>
      <tr key={i}>
        <td>{o.prize}</td>
        <td>{dateFormat(o.entry_deadline, "fullDate")}</td>
        <td>{dateFormat(o.drawing_date, "fullDate")}</td>
      </tr>
    )

    return (
        <div class="drawing_sched_lg">
          <h2 class="drawing_schedule_title">Drawing Schedule</h2>
          <table>
            <tbody>
              <tr>
                <th>Prize</th>
                <th>Entry Deadline</th>
                <th>Drawing Date</th>
              </tr>
              {rows}
            </tbody>
          </table>
        </div>
    );
  }
}
