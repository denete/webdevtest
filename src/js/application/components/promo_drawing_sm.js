import React from "react";
import { Link } from "react-router";
import dateFormat from "dateformat";

// small format for drawing schedule table
export default class PromoDrawingSm extends React.Component {
  render(){
    const drawings = this.props.drawings;

    const rows = drawings.map((o,i)=>
      <table key={i}>
        <tbody>
          <tr>
            <th>Prize</th>
            <td>{o.prize}</td>
          </tr>
          <tr>
            <th>Entry Deadline</th>
            <td>{dateFormat(o.entry_deadline, "fullDate")}</td>
          </tr>
          <tr>
            <th>Drawing Date</th>
            <td>{dateFormat(o.drawing_date, "fullDate")}</td>
          </tr>
        </tbody>
      </table>
    )

    return (
        <div class="drawing_sched_sm">
          <h2 class="drawing_schedule_title">Drawing Schedule</h2>
          {rows}
        </div>
    );
  }
}
