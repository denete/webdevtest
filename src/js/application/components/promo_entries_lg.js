import React from "react";
import { Link } from "react-router";
import dateFormat from "dateformat";

// table for large format Ticket entries
export default class PromoEntriesLg extends React.Component {
  render(){
    const entries = this.props.entries;

    const rows = entries.map((o,i)=>
      <tr key={i}>
        <td>{o.entry_number}</td>
        <td>{dateFormat(o.date, "fullDate")}</td>
      </tr>
    )

    return (
      <div class="ticket_entries_lg">
        <h2 class="total_entered">Your total Tickets Entered: {entries.length}</h2>
        <p>All Entries are locked in at the time they are submitted and cannotbe deleted.</p>
        <table>
          <thead>
            <tr>
              <th>Entry Number</th>
              <th>Date</th>
            </tr>
          </thead>
          <tbody>
            {rows}
          </tbody>
        </table>
      </div>
    );
  }
}
