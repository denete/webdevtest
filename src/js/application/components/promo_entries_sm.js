import React from "react";
import { Link } from "react-router";
import dateFormat from "dateformat";

// table for small format Ticket entries
export default class PromoEntriesSm extends React.Component {
  render(){
    const entries = this.props.entries;

    const rows = entries.map((o,i)=>
      <table key={i}>
        <tbody>
          <tr>
            <th>Entry Number</th>
            <td>{o.entry_number}</td>
          </tr>
          <tr>
            <th>Date</th>
            <td>{dateFormat(o.date, "fullDate")}</td>
          </tr>
        </tbody>
      </table>
    )

    return (
      <div class="ticket_entries_sm">
        <h2 class="total_entered">Your total Tickets Entered: {entries.length}</h2>
        <p>All Entries are locked in at the time they are submitted and cannotbe deleted.</p>
        {rows}
      </div>
    );
  }
}
