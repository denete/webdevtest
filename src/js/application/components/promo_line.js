import React from "react";
import { Link } from "react-router";
import dateFormat from "dateformat";

// Each promotion line
export default class PromoLine extends React.Component {

  render(){
    const promo_item = this.props.promo_item;
    const promo_index = this.props.promo_index;

    return (
      <div class="promo_line">
        <div class="">
          <img src={"./" + promo_item.promo_image_url} alt=""/>
          <Link to={"/promotion/" + promo_index}><h2>{promo_item.promotion_name}</h2></Link>
          <p>{promo_item.summary}</p>
          <p>Next Drawing Date: {dateFormat(promo_item.drawings[0].drawing_date, "fullDate")} </p>
        </div>
      </div>
    );
  }
}
