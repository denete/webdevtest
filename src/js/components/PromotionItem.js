import React from "react";

import moment from "moment";

import { injectIntl, defineMessages } from "react-intl";

import { Table, Thead, Tbody, Tr, Th, Td } from "react-super-responsive-table";

import "react-super-responsive-table/dist/SuperResponsiveTableStyle.css";

const messages = defineMessages({
	drawing_schedule: {
		id: "DRAWING_SCHEDULE",
		defaultMessage: "Drawings schedule"
	},
	total_tickets_entered: {
		id: "TOTAL_TICKETS_ENTERED",
		defaultMessage: "Your total tickets entered"
	},
	entry_description: {
		id: "ENTRY_DESCRIPTION",
		defaultMessage: "All entries are locked in at the time they are submitted and cannot be deleted."
	}
});

const drawingHeaders = ["Prize", "Entry deadline", "drawing date"];

const entryHeaders = ["Entry number", "date"];

class PromotionItem extends React.Component {
    renderDrawingSchedule (promotionData) {
        const { intl } = this.props;

        const drawingsTHeadContent = drawingHeaders.map((drawingHeader, index) => {
            const headerKey = `header${index}`;
            return <Th key={ headerKey }>{ drawingHeader }</Th>;
        });

        const drawingsTBodyContent = promotionData.drawings.map((drawingData, index) => {
            const key = `drawings${index}`;
            const deadlineDate = moment(drawingData.entry_deadline).format("dddd, MMMM DD, YYYY");
            const drawingDate = moment(drawingData.drawing_date).format("dddd, MMMM DD, YYYY");
            return (
                <Tr key={key}>
                    <Td>{drawingData.prize}</Td>
                    <Td>{deadlineDate}</Td>
                    <Td>{drawingDate}</Td>
                </Tr>
            );
        });

        const drawingScheduleText = intl.formatMessage(messages.drawing_schedule);
        return (
            <div>
                <p>{ drawingScheduleText }</p>
                <Table className="tableResults">
                    <Thead>
                        <Tr>{ drawingsTHeadContent }</Tr>
                    </Thead>

                    <Tbody>
                        { drawingsTBodyContent }
                    </Tbody>
                </Table>
            </div>
        );
    }

    renderEntries (promotionData) {
        const { intl } = this.props;

        const entriesTHeadContent = entryHeaders.map((entryHeader, index) => {
            const headerKey = `header${index}`;
            return <Th key={ headerKey }>{ entryHeader }</Th>;
        });

        const entriesTBodyContent = promotionData.entries.map((entryData, index) => {
            const key = `entries${index}`;
            const entryDate = moment(entryData.date).format("dddd, MMMM DD, YYYY");
            return (
                <Tr key={key}>
                    <Td>{entryData.entry_number}</Td>
                    <Td>{entryDate}</Td>
                </Tr>
            );
        });

        const totalTicketsEnteredText = intl.formatMessage(messages.total_tickets_entered);
        const entryDescriptionText = intl.formatMessage(messages.entry_description);
        return (
            <div>
                <p>{ totalTicketsEnteredText } { promotionData.entries.length }</p>
                <p>{ entryDescriptionText }</p>
                <Table className="tableResults">
                    <Thead>
                        <Tr>
                            { entriesTHeadContent }
                        </Tr>
                    </Thead>

                    <Tbody>
                        { entriesTBodyContent }
                    </Tbody>
                </Table>
            </div>
        );
    }

    render () {
        const { promotionData } = this.props;

        return (
            <div>
                <img src={promotionData.promo_image_url}></img>
                <div>{promotionData.promotion_name}</div>
                { this.renderDrawingSchedule(promotionData) }
                <p>{promotionData.entry_info}</p>
                { this.renderEntries(promotionData) }
            </div>
        );
    }
}

export default injectIntl(PromotionItem);