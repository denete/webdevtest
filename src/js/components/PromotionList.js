import React from "react";
import numeral from "numeral";
import PromotionListItem from "./PromotionListItem";

class PromotionList extends React.Component {
    render () {
        const { promotion_objects: promotionsData, server_time: serverTimeData } = this.props.data;
        const promotionListItems = promotionsData.map((promotionData, index) => {
            const key = `promotionListItem${index}`;
            const url = `?promo=promo${numeral(index + 1).format("00")}`;
            return (
                <PromotionListItem
                    key={key}
                    url={url}
                    serverTimeData={serverTimeData}
                    promotionData={promotionData}
                />
            );
        });
        return (
            <div className="promotionListView">
                <div className="promotionList">
                    { promotionListItems }
                </div>
            </div>
        )
    }
}

export default PromotionList;