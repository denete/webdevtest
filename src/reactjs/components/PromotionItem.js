import React from "react";

import moment from "moment";

import { injectIntl, defineMessages } from "react-intl";

import { nextEventInTime } from "./../utils/nextEventInTime";

import ErrorComponent from "./ErrorComponent";

import DrawingsTable from "./DrawingsTable";

import EntriesTable from "./EntriesTable";

const messages = defineMessages({
    next_entry_deadline: {
        id: "NEXT_ENTRY_DEADLINE",
        defaultMessage: "The Next Entry Deadline is"
    },
    date: {
        id: "DATE",
        defaultMessage: "Date"
    }
});

export class PromotionItem extends React.Component {

    render () {
        const { data, promotionIndex, intl } = this.props;
        const promotionData = data.promotion_objects[promotionIndex];
        if (promotionData === undefined) {
            return <ErrorComponent />;
        }
        const serverTimeData = data.server_time;

        const nextEntryDeadlineText = intl.formatMessage(messages.next_entry_deadline);
        const nextEntryData = nextEventInTime(promotionData.entries, "date", serverTimeData);
        const nextEntryDate = moment(nextEntryData.date).format("dddd, MMMM DD, YYYY");

        const imageSrc = `/${promotionData.promo_image_url}`;
        return (
            <div className="promotionView">
                <div className="promotionItem container">
                    <div className="nextEntryContainer">
                        <p>{ nextEntryDeadlineText }</p>
                        <p>{ nextEntryDate }</p>
                    </div>
                    <div className="promotionItemImg">
                        <img src={ imageSrc }></img>
                    </div>
                    <h1>{ promotionData.promotion_name }</h1>
                    <p>{ promotionData.summary }</p>
                    <DrawingsTable promotionData={ promotionData } />
                    <p>{ promotionData.entry_info }</p>
                    <EntriesTable promotionData={ promotionData } />
                </div>
            </div>
        );
    }
}

export default injectIntl(PromotionItem);