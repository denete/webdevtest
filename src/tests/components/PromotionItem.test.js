import React from 'react';
import { shallow } from 'enzyme';
import { nodeWithIntlProp } from "./../IntlTestHelper";
import { PromotionItem } from "./../../reactjs/components/PromotionItem";
import testData from "./../testData";

test('should render PromotionItem correctly', () => {
    const wrapper = shallow(nodeWithIntlProp(<PromotionItem data={testData} promotionIndex={0} />));
    expect(wrapper).toMatchSnapshot();
});