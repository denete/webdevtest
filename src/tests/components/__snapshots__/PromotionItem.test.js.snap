// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PromotionItem correctly 1`] = `
<div
  className="promotionView"
>
  <div
    className="promotionItem container"
  >
    <div
      className="nextEntryContainer"
    >
      <p>
        The Next Entry Deadline is
      </p>
      <p>
        Tuesday, June 07, 2016
      </p>
    </div>
    <div
      className="promotionItemImg"
    >
      <img
        src="images/fortune.jpg"
      />
    </div>
    <h1>
      $1,000,000 Fortune Second-Chance Promotion
    </h1>
    <p>
      Enter for the chance to win the Second-Chance prize of $1,000,000 in the $1,000,000 Fortune Promotion!
    </p>
    <div
      className="drawingScheduleTable"
    >
      <h2>
        Drawing Schedule
      </h2>
      <Table
        className="tableResults"
      >
        <Thead>
          <WithContext(TrInner)>
            <Th
              key="header0"
            >
              Prize
            </Th>
            <Th
              key="header1"
            >
              Entry Deadline
            </Th>
            <Th
              key="header2"
            >
              Drawing Date
            </Th>
          </WithContext(TrInner)>
        </Thead>
        <Tbody>
          <WithContext(TrInner)
            key="drawings0"
          >
            <WithContext(TdInner)>
              $10,000
            </WithContext(TdInner)>
            <WithContext(TdInner)>
              Tuesday, June 07, 2016
            </WithContext(TdInner)>
            <WithContext(TdInner)>
              Monday, June 13, 2016
            </WithContext(TdInner)>
          </WithContext(TrInner)>
          <WithContext(TrInner)
            key="drawings1"
          >
            <WithContext(TdInner)>
              $1,000,000
            </WithContext(TdInner)>
            <WithContext(TdInner)>
              Monday, June 20, 2016
            </WithContext(TdInner)>
            <WithContext(TdInner)>
              Sunday, June 26, 2016
            </WithContext(TdInner)>
          </WithContext(TrInner)>
        </Tbody>
      </Table>
    </div>
    <p>
      Any non-winning $5 ticket is eligible for entry. Each ticket submitted will result in one (1) entry into the drawing. An unlimited number of qualifying tickets can be entered, but each ticket can only be entered once. You do not need to keep original tickets to claim a second-chance prize. However, once Official Entry Codes have been submitted online, please check the regular drawing results before discarding tickets.
    </p>
    <div
      className="entriesTable"
    >
      <h2>
        Your Total Tickets Entered:
         
        3
      </h2>
      <p
        className="lockedEntry"
      >
        All entries are locked in at the time they are submitted and cannot be deleted.
      </p>
      <Table
        className="tableResults"
      >
        <Thead>
          <WithContext(TrInner)>
            <Th
              key="header0"
            >
              Entry number
            </Th>
            <Th
              key="header1"
            >
              Date
            </Th>
          </WithContext(TrInner)>
        </Thead>
        <Tbody>
          <WithContext(TrInner)
            key="entries0"
          >
            <WithContext(TdInner)>
              027840217109
            </WithContext(TdInner)>
            <WithContext(TdInner)>
              Tuesday, June 07, 2016
            </WithContext(TdInner)>
          </WithContext(TrInner)>
          <WithContext(TrInner)
            key="entries1"
          >
            <WithContext(TdInner)>
              187554635107
            </WithContext(TdInner)>
            <WithContext(TdInner)>
              Wednesday, June 08, 2016
            </WithContext(TdInner)>
          </WithContext(TrInner)>
          <WithContext(TrInner)
            key="entries2"
          >
            <WithContext(TdInner)>
              314725611824
            </WithContext(TdInner)>
            <WithContext(TdInner)>
              Thursday, June 09, 2016
            </WithContext(TdInner)>
          </WithContext(TrInner)>
        </Tbody>
      </Table>
    </div>
  </div>
</div>
`;
